src/c++/perf_analyzer/inference_profiler.cc

```diff
@@ -638,8 +651,41 @@ InferenceProfiler::ProfileHelper(
           *is_stable = false;
         }
       }
-      if (*is_stable) {
-        break;
+      if (IsMPIRun()) {
+        int world_size{1};
+        MPI_Comm_size(MPI_COMM_WORLD, &world_size);
```

Comment:
The below stability check logic looks good to me. But you must move this into a new function for better readability.

Reply:
moved to new function 👍